skyportal/handlers/api/internal/plot.py

```diff
@@ -117,6 +122,52 @@ def get(self, obj_id, telescope_id):
         if sunset > sunrise:
             sunset = telescope.observer.sun_set_time(time, which='previous')
 
-        json = self.calculate_airmass(obj, telescope, sunrise, sunset)
+        json = self.calculate_airmass(obj, telescope, sunrise, sunset).to_dict(
+            orient='records'
+        )
+        self.verify_and_commit()
+        return self.success(data=json)
+
+
+class PlotHoursBelowAirmassHandler(AirmassHandler):
+    @auth_or_token
+    def get(self, obj_id, telescope_id):
+        threshold = cfg["misc.hours_below_airmass_threshold"]
+        if threshold is None:
+            threshold = 2.9
+
+        obj = Obj.get_if_accessible_by(obj_id, self.current_user, raise_if_none=True)
+        try:
+            telescope_id = int(telescope_id)
+        except TypeError:
+            return self.error(f'Invalid telescope id: {telescope_id}, must be integer.')
+
+        telescope = Telescope.get_if_accessible_by(
+            telescope_id, self.current_user, raise_if_none=True
+        )
+
+        year = datetime.date.today().year
+        year_start = datetime.datetime(year, 1, 1, 0, 0, 0)
+
+        json = []
+        # Sample every 7 days for the year
+        for day in range(52):
```

Comment:
Maybe best to make 7 a variable (deltat) and then do ceil(365/deltat)?

Reply:
Good suggestion, done